src/ducks/apps/index.js

```diff
@@ -268,10 +268,12 @@ export function fetchIconsProgressively() {
   return async (dispatch, getState) => {
     const apps = getState().apps.list
     const iconsMap = new Map() // slug: icon
-    apps.forEach(async (app, index) => {
+    let updateCounter = 0
+    apps.forEach(async app => {
       const iconData = await _getIcon(app.iconToLoad)
       iconsMap.set(app.slug, iconData)
-      if (iconsMap.size % 10 === 0 || index === apps.length - 1) {
+      updateCounter++
+      if (iconsMap.size % 10 === 0 || updateCounter === apps.length) {
         dispatch({ type: RECEIVE_APPS_ICON, iconsMap })
```

Comment:
Here, if the update were asynchronous, you would have a problem since the map would be empty when received in the reducer. I think you should dispatch a copy of the map.

Reply:
Indeed, I may need to improve the map handling a little here 👍